### PR TITLE
WIP: Strip out unchanged keys on object update

### PR DESCRIPTION
**Pull request by dblythy**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-Swift/security/policy).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-Swift/issues?q=is%3Aissue).

### Issue Description

Draft PR for ideas / suggestions for improvements around this approach.

If this isn't suitable for this project, I'll be happy to work on this branch for my own usage. I would welcome feedback so I can make sure my fork is as good as it can be 😊 

Related issue: #242

### Approach

Stores objects in JSON form in memory. When objects are saved, adds unchanged keys to SkipKeys.

### TODOs before merging

- [ ] Add tests
- [ ] Add entry to changelog

### CODE REVIEW

File: Sources/ParseSwift/Storage/JSONStorage.swift [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        guard let objectable = object as? Objectable else {
+            return keysToSkip
+        }
+        do {
+            let encoded = try ParseCoding.parseEncoder().encode(object)
+            if let id = objectable.objectId,
+               let previousStore = store[id],
+               let json = try JSONSerialization.jsonObject(with: encoded, options: []) as? [String: AnyObject] {
+                for keyName in json.keys where
+                (json[keyName] as? NSObject) == (previousStore[keyName] as? NSObject)
```
Comment (cbaker6) [inline Sources/ParseSwift/Storage/JSONStorage.swift:44]: Will need to remove casting to `NSObject` as it doesn't work well with this SDK. See #274 and #275 for workarounds.

**Review by cbaker6 — CHANGES_REQUESTED** [review]

**Comment by mtrezza** [comment]
Re-opened as this PR suggests a improvements tracked in https://github.com/parse-community/Parse-Swift/issues/401

**Comment by cbaker6** [comment]
Closing due to inactivity since 10/21. @dblythy feel free to reopen if you can address the issues mentioned in https://github.com/parse-community/Parse-Swift/pull/263#issuecomment-954269490, https://github.com/parse-community/Parse-Swift/pull/263#issuecomment-954306992, and https://github.com/parse-community/Parse-Swift/pull/263#discussion_r742961428 and if you think the contribution of this PR will be an improvement over #406

**Comment by mtrezza** [comment]
Since you closed this, we'll need to open another issue to track this discussion about further improvement.

**Comment by dblythy** [comment]
Yes - sorry - I closed this as hearing the valid comments from @Vortec4800, I decided this approach probably wasn't the best way to go. I might reconsider / retry having a crack again in the future

**Pull request closed without merging**